tests/unit/storage_v2_replication.cpp

```diff
@@ -689,23 +693,110 @@ TEST_F(ReplicationTest, ReplicationReplicaWithExistingEndPoint) {
 
-  const std::string replica1_name{"REPLICA1"};
+  const std::string replica1_name{replicas[0]};
   ASSERT_FALSE(main_store
                    .RegisterReplica(replica1_name, replica1_endpoint,
                                     memgraph::storage::replication::ReplicationMode::SYNC, {.timeout = 2.0})
                    .HasError());
 
-  const std::string replica2_name{"REPLICA2"};
+  const std::string replica2_name{replicas[1]};
   ASSERT_TRUE(
       main_store
           .RegisterReplica(replica2_name, replica2_endpoint, memgraph::storage::replication::ReplicationMode::ASYNC)
           .GetError() == memgraph::storage::Storage::RegisterReplicaError::END_POINT_EXISTS);
 }
+
+TEST_F(ReplicationTest, RestoringReplicationAtStartupAftgerDroppingReplica) {
+  auto main_config = configuration;
+  main_config.durability.restore_replicas_on_startup = true;
+  auto main_store = std::make_unique<memgraph::storage::Storage>(main_config);
+
+  memgraph::storage::Storage replica_store1(configuration);
+  replica_store1.SetReplicaRole(memgraph::io::network::Endpoint{local_host, ports[0]});
+
+  memgraph::storage::Storage replica_store2(configuration);
+  replica_store2.SetReplicaRole(memgraph::io::network::Endpoint{local_host, ports[1]});
+
+  auto res = main_store->RegisterReplica(replicas[0], memgraph::io::network::Endpoint{local_host, ports[0]},
+                                         memgraph::storage::replication::ReplicationMode::SYNC);
+  ASSERT_FALSE(res.HasError());
+  res = main_store->RegisterReplica(replicas[1], memgraph::io::network::Endpoint{local_host, ports[1]},
+                                    memgraph::storage::replication::ReplicationMode::SYNC);
+  ASSERT_FALSE(res.HasError());
+
+  auto replica_infos = main_store->ReplicasInfo();
+
+  ASSERT_EQ(replica_infos.size(), 2);
+  ASSERT_EQ(replica_infos[0].name, replicas[0]);
+  ASSERT_EQ(replica_infos[0].endpoint.address, local_host);
+  ASSERT_EQ(replica_infos[0].endpoint.port, ports[0]);
+  ASSERT_EQ(replica_infos[1].name, replicas[1]);
+  ASSERT_EQ(replica_infos[1].endpoint.address, local_host);
+  ASSERT_EQ(replica_infos[1].endpoint.port, ports[1]);
+
+  main_store.reset();
+
+  auto other_main_store = std::make_unique<memgraph::storage::Storage>(main_config);
+  replica_infos = other_main_store->ReplicasInfo();
+  ASSERT_EQ(replica_infos.size(), 2);
+  ASSERT_EQ(replica_infos[0].name, replicas[0]);
+  ASSERT_EQ(replica_infos[0].endpoint.address, local_host);
+  ASSERT_EQ(replica_infos[0].endpoint.port, ports[0]);
+  ASSERT_EQ(replica_infos[1].name, replicas[1]);
+  ASSERT_EQ(replica_infos[1].endpoint.address, local_host);
+  ASSERT_EQ(replica_infos[1].endpoint.port, ports[1]);
+}
+
+TEST_F(ReplicationTest, RestoringReplicationAtStartup) {
+  auto main_config = configuration;
+  main_config.durability.restore_replicas_on_startup = true;
+  auto main_store = std::make_unique<memgraph::storage::Storage>(main_config);
+  memgraph::storage::Storage replica_store1(configuration);
```

Comment:
You don't have to act on that, as this is functionality is already there, I just would like to raise the attention here: the two storages are using the same data directory and the lock file doesn't fail because the two storage are in the same process. Thus it might cause some issues in the future, so it is good if more people is aware of this.

Reply:
Good remark, I think it's good to know for everybody. That could explain some weird behavior in unit test.
Remark regarding the duplication of config however: here the main_Memgraph wants to allow `restore_replicas_on_startup`, whereas we do not want to have it enable for the replicas themselves (attribute `restore_replicas_on_startup` is `false` by default)